cmd/queue/main.go

```diff
@@ -155,9 +155,21 @@ func proxyHandler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, tracingEn
 
 		// Enforce queuing and concurrency limits.
 		if breaker != nil {
+			var cf func()
+			if tracingEnabled {
+				ctx, waitSpan := trace.StartSpan(r.Context(), "queueWait")
+				r = r.WithContext(ctx)
+				cf = waitSpan.End
+			}
 			if err := breaker.Maybe(r.Context(), func() {
+				if cf != nil {
+					cf()
+				}
 				next.ServeHTTP(w, r)
 			}); err != nil {
+				if cf != nil {
+					cf()
+				}
```

Comment:
This is really not pretty. Defer should be used.

Reply:
Outdated, but still stands

Reply:
Defer does not work here, since we need to terminate as soon as queueing is finished (whether successfully or not). There's no way we can wrap this in a func as well (because of how Try interface works).